### PR TITLE
Update directory targeting logic to support glob patterns

### PR DESCRIPTION
**Pull request by mozex**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Fixed tickets | #... 

This pull request updates the directory targeting logic to support glob patterns. Previously, only specific directories could be used with the `->in()` method. With this update, it's now possible to use glob patterns to specify directories.

### Changes
Replaced is_dir() and file_exists() checks with glob() in the function responsible for collecting the test files. This allows for directory paths with wildcard characters, enabling glob pattern matching.
This change makes the test selection process more flexible and intuitive. For example, it enables pattern-based targeting in test usage like this:

```php
uses(
    DuskTestCase::class,
    DatabaseMigrations::class
)->in('../domain/*/tests/Browser');

uses(
    TestCase::class,
    RefreshDatabase::class
)->in('../domain/*/tests/[!Browser]*');
```

### Benefits
The main benefits of this change are:

More flexibility in selecting test directories.
Using glob patterns allows for broader or more specific targeting of test cases.
It simplifies adding and managing tests, especially in larger projects.

### Testing
All existing tests pass with the new changes. Please consider this pull request for merging.

### CODE REVIEW

**Comment by fabio-ivona** [comment]
Lgtm, would you mind to add some tests for this feature?

**Comment by mozex** [comment]
> Lgtm, would you mind to add some tests for this feature?

Sure, working on it.

**Comment by mozex** [comment]
Let me know if you need me to write more tests.

**Comment by fabio-ivona** [comment]
@mozex would you mind to check failing tests?

**Comment by mozex** [comment]
> @mozex would you mind to check failing tests? 

I am looking into it

**Comment by mozex** [comment]
fixed both tests and phpstan error

File: src/PendingCalls/UsesCall.php [hunk 1]
```diff
@@ -84,8 +84,10 @@ public function in(string ...$targets): void
         }, $targets);
 
         $this->targets = array_reduce($targets, function (array $accumulator, string $target): array {
-            if (is_dir($target) || file_exists($target)) {
-                $accumulator[] = (string) realpath($target);
+            if (($matches = glob($target)) != false) {
```
Comment (owenvoke) [inline src/PendingCalls/UsesCall.php]: Perhaps use strict equality here?
```suggestion
            if (($matches = glob($target)) !== false) {
```

File: tests/Pest.php [hunk 2]
```diff
@@ -1,9 +1,14 @@
 <?php
 
+use Tests\CustomTestCase\CustomTestCase;
 use Tests\CustomTestCaseInSubFolders\SubFolder\SubFolder\CustomTestCaseInSubFolder;
 
 uses(CustomTestCaseInSubFolder::class)->in('PHPUnit/CustomTestCaseInSubFolders/SubFolder/SubFolder');
 
+uses(CustomTestCase::class)->in('PHPUnit/CustomTestCase/SubFolder/*/');
```
Comment (owenvoke) [inline tests/Pest.php]: I feel like maybe these should be somewhere else with a more logical name (something like, `GlobPatternTests`) just so we know why the tests are there.

Perhaps add a comment to the test cases too?

Comment (mozex) [reply tests/Pest.php]: sure, I will make the changes soon

**Review by owenvoke — APPROVED** [review]

**Review by fabio-ivona — APPROVED** [review]

**Comment by mozex** [comment]
When do you think this will be merged?

**Comment by nunomaduro** [comment]
@mozex Can you pull request docs on this? https://github.com/pestphp/docs/edit/master/configuring-tests.md

**Comment by mozex** [comment]
> @mozex Can you pull request docs on this? https://github.com/pestphp/docs/edit/master/configuring-tests.md

Sure, i will send a PR today.

**Pull request merged**